### PR TITLE
Added player state and player id for clients, updated gitignore also

### PR DESCRIPTION
**Pull request by Nipunwahi**
Added a new type called player state and added player id also.

Note - Updated gitignore to ignore stack.yaml.lock file

### CODE REVIEW

File: src/Game.hs [hunk 1]
```diff
@@ -32,6 +30,7 @@ instance Named Team where
 
 data Gamestate = Gamestate {remainingCards :: [Card], round::Int, playerOrder::[Player], currentRound::[Card], teams::(Team,Team), trump::Suit} deriving (Show,Read)
 
+data PlayerState = PlayerState {playerName :: String, playerId::String, playerCards:: [Card], currentRoundCard :: [Card], roundNumber::Int, teamInfo::(Team,Team), trumpSuit:: Suit} deriving (Show,Read)
```
Comment (MaheshBharadwaj) [inline src/Game.hs]:
```suggestion
data PlayerState = PlayerState {player :: Player, currentRoundCard :: [Card], roundNumber::Int, teamInfo::(Team,Team), trumpSuit:: Suit} deriving (Show,Read)
```

File: src/Game.hs [hunk 2]
```diff
@@ -97,17 +96,20 @@ assignCardPl numCards currentState@(remCards,p:ps)
...
 suresh :: Player
-suresh = Player "Suresh" []
+suresh = Player "Suresh" "4" []
 
 allPlayers :: [Player]
 allPlayers = [nipun,anish,mahesh,suresh]
 
+-- PlayerState to send to the client of player for displaying on UI
+getPlayerState :: Gamestate -> Player -> PlayerState
+getPlayerState gs player = PlayerState (name player) (_id player) (cards player) (currentRound gs) (round gs) (teams gs) (trump gs)
```
Comment (MaheshBharadwaj) [inline src/Game.hs]:
```suggestion
getPlayerState gs player = PlayerState player (currentRound gs) (round gs) (teams gs) (trump gs)
```

**Review by MaheshBharadwaj — CHANGES_REQUESTED** [review]
Why can't we reuse Player in PlayerState?

File: src/Game.hs (hunk 2 above)
Comment (Nipunwahi) [reply src/Game.hs]: yeah we can do that, missed it, would commit the suggestion

**Review by MaheshBharadwaj — APPROVED** [review]

**Pull request merged**